tests/Rules.Framework.IntegrationTests/Tests/Scenario5/BestServerTests.cs

```diff
@@ -0,0 +1,289 @@
+namespace Rules.Framework.IntegrationTests.Tests.Scenario5
+{
```

Comment:
Consider having a Feature test instead of a Scenario test, to test the behaviour added in this PR.
Feature tests are simpler and help the developer to understand what exact feature is meant to be tested with that test.

Reply:
Done